### PR TITLE
Remove constant respawn reads, save respawn pos entering bed

### PR DESCRIPTION
**Pull request by tenplus1**
Save respawn position when entering bed. I also removed the constant beds.read_spawns() calls when a player joins as this is only required once

### CODE REVIEW

**Comment by paramat** [comment]
Thanks for combinig PRs, that was needed.

**Comment by paramat** [comment]
Does the commit message need editing? I can't see code changes for items 1 and 4.

**Comment by tenplus1** [comment]
@paramat nope seems fine as is.

**Comment by sofar** [comment]
The commit message is indeed still wrong - don't look at the top PR comment :)

**Comment by tenplus1** [comment]
@sofar I dont know how to change commit message, never use git console, only the web interface...

**Comment by sofar** [comment]
`git commit --amend`

**Comment by tenplus1** [comment]
Please don't let this pull stagnate because of a commit title, it's a necessary fix.

**Comment by sofar** [comment]
```
mkdir temp
cd temp
git clone git@github.com:tenplius1/minetest_game
cd minetest_game
git checkout patch-39
git commit --amend
# edit the commit message
git push origin patch-39:patch-39 -f
```

It's not OK to ask us to commit a change with a bad commit message. A proper commit should include a proper commit message. In this case, the message is clearly wrong on 2 of the 4 items it claims to implement.

**Comment by paramat** [comment]
We mean this is wrong, this is the actual commit message used in history, the thread title isn't used.

- Add message to chat when players sleep in a bed
- Save respawn position when entering bed
- Remove constant beds.read_spawns() calls when a player joins as this is only required once (separate pull)
- Remove sleeping message (prevent spam)

Items 1 and 4 are not part of this PR correct?
Anyway we can easily alter the commit message when merging so don't worry about doing this.

**Comment by paramat** [comment]
You're a valued contributor so please learn how to fork, branch and submit properly, it will be easier and quicker for you and you can work onn your own. The bootcamp takes you through step by step and gives you all the console commands to use:

Bottom of github page > 'help' > 'bootcamp'
'set up git' if you need to
'fork a repo'
then > 'syncing a fork'.
then > 'branches' then > 'create a pull request'

**Comment by paramat** [comment]
Now reviewing. Please can you explain this: "Save respawn position when entering bed". I guess it is to save respawn positions without having to have all players in bed and the night skipped? It speeds up review if you assume the dev doesn't know much about the mod (which i didn't until studying it now).

You've moved it into ```update_formspecs()``` which is potentially called twice and only called if 'not is_sp' so respawn positions are not saved in singleplayer?
Perhaps it should be moved to somewhere near the end of ```beds.on_rightclick(pos, player)```

File: mods/beds/functions.lua [hunk 1]
```diff
@@ -111,7 +111,9 @@ local function update_formspecs(finished)
 		if is_majority and is_night_skip_enabled() then
 			form_n = form_n .. "button_exit[2,8;4,0.75;force;Force night skip]"
 		end
-	end
+
+		beds.set_spawns() -- save respawn positions when entering bed
+end
```
Comment (paramat) [inline mods/beds/functions.lua]: Needs iindent.

**Comment by paramat** [comment]
Added line comment about 'end' needing indent.

**Comment by paramat** [comment]
How about this, to avoid setting spawn when standing up? (lay_down(player, nil, nil, false) is stand up)
```
	-- move to bed
	if not beds.player[name] then
		lay_down(player, ppos, pos)
                beds.set_spawns() -- save respawn positions when entering bed
	else
		lay_down(player, nil, nil, false)
end
```

**Comment by paramat** [comment]
:+1: Seems good but need review from others.

**Comment by paramat** [comment]
:+1: from nore http://irc.minetest.ru/minetest-dev/2016-07-13#i_4649569

**Comment by paramat** [comment]
d7a74560072b0e2c2b0f35b49cbd46a0a1852a4a

**Pull request closed without merging**